### ISSUE TITLE
SFTTrainer loss value decreasing only for certain value of max_seq_length

### ISSUE THREAD

**Issue by Jeevi10** [huggingface/trl#940]
trainer = SFTTrainer(
    model=model,
    train_dataset= dataset['train'],
    peft_config=peft_config,
    dataset_text_field='text',
    max_seq_length=max_seq_length,
    tokenizer=tokenizer,
    args=training_arguments,
  
)

0%|          | 0/160 [00:00<?, ?it/s]
 1%|          | 1/160 [00:08<21:09,  7.98s/it]
{'loss': 2.0412, 'learning_rate': 4e-05, 'epoch': 0.01}
1%|▏         | 2/160 [04:33<7:00:02, 159.51s/it]
{'loss': 0.0, 'learning_rate': 8e-05, 'epoch': 0.03}
 When I start training I see loss value and its become zero for second step, also its weird if I decrease the  max_seq_value to certain value the loss value is decreasing ? can someone explain this to me  ?

**Comment by YeonwooSung** [comment]
I also had some similar issue. For me, the problem was that I appended invalid DataCollator, which makes the SFTTrainer to ignore to calculate the loss value.
